### ISSUE TITLE
usePrepareContractWrite not working as expected

### ISSUE THREAD

**Issue by 0x62** [unicape/use-wagmi#8]
I'm trying to use `usePrepareContractWrite` and `useContractWrite` as follows:

```
const { config } = usePrepareContractWrite({
  ...contractStore.wagmiConfig, // address, abi
  functionName: 'myFn',
  args: [props.address],
  onSettled(data, error) {
    console.log('Prepare settled', { data, error })
  },
})

const { isLoading, write } = useContractWrite(config)

const onClick = () => {
  console.log(config, write)
}
```

`onSettled` is called with success state, and I can see the `request` key has been prepared. However when the `onClick` handler is called, `config.request` is still undefined, as is the write function.

**Comment by unicape** [comment]
@0x62 Yes, I know this problem, the reason is because config is not responsive, I will fix this problem in the next beta version

**Comment by unicape** [comment]
@0x62 A new version has been released. Please use the latest version. There are some changes in the API. Please check the README or example. This issue will be closed first.

**Comment by 0x62** [comment]
Thanks, confirmed working
